### PR TITLE
Footer consistency all over app

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
#### What are the relevant tickets?
- fixes https://github.com/mitodl/micromasters/issues/1489

#### What's this PR do?
- Made footer consistent all over app
 - [x] Dashboard and user pages 
 - [x] Search
 - [x] Program
 - [x] Terms of service

#### Where should the reviewer start?
- See ui/templates/dashboard.html

#### How should this be manually tested?
- see pages
 - [x] Dashboard and user pages 
 - [x] Search
 - [x] Program
 - [x] Terms of service

@pdpinch @aliceriot @noisecapella 

#### Screenshots (if appropriate)
screen shot 2016-10-24 at 3 05 38 pm

### CODE REVIEW

**Comment by Ferdi** [comment]
Thanks @amir-qayyum-khan . If page is behind login,  sharing does not make sense, right ?  /cc @roberthouse54

**Comment by amir-qayyum-khan** [comment]
@Ferdi what do you mean by ```behind login,```?
Do you mean user on logout state on home page?

**Comment by roberthouse54** [comment]
I'd say that is correct. So the only pages that should have the share icons are the marketing pages. I.e. The micromasters home page, and the program pages.

File: ui/templates/dashboard.html [hunk 1]
```diff
@@ -5,5 +5,7 @@
 
 {% block content %}
 <div id="dashboard"></div>
+{% include "footer.html" %}
 <script type="text/javascript" src="{{ dashboard_src }}"></script>
+<script type="text/javascript" src="{{ public_src }}"></script>
```
Comment (aliceriot) [inline ui/templates/dashboard.html]: what do we need `public_src` on the dashboard for?

Comment (amir-qayyum-khan) [inline ui/templates/dashboard.html]: this for social code, there are social buttons for sharing.

**Comment by amir-qayyum-khan** [comment]
removed footer from 
- Dashboard and user pages
- Search
- Terms of service

**Comment by pdpinch** [comment]
@amir-qayyum-khan I think you removed too much. Can you include a screenshot of the homepage and the learner dashboard? 

@roberthouse54 @Ferdi I understand why we don't want the sharing buttons on the learner dashboard, but what about the MIT logo, copyright message, etc?

**Comment by Ferdi** [comment]
I'm confused what we are trying to solve here. From what I understand

1 - In all pages, the footer should look consistent  like https://goo.gl/Agv6ju
2 - Add social icons in public pages only

**Comment by roberthouse54** [comment]
Ferdi's comment above is correct. We should use the same footer component on all pages, and basically hide the share links on all pages except the public pages.

**Comment by Ferdi** [comment]
@roberthouse54  should "Give to MIT" be left or right. If right,  how does it look when social icons are present ?

**Comment by roberthouse54** [comment]
Left.

**Comment by amir-qayyum-khan** [comment]
@pdpinch @roberthouse54 @Ferdi 

#### layout on marketing pages
screen shot 2016-10-25 at 1 19 54 pm

#### layout on non marketing pages
screen shot 2016-10-25 at 1 19 20 pm

**Pull request merged**